src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/RSACng.ImportExport.cs

```diff
@@ -38,10 +38,13 @@ public override void ImportParameters(RSAParameters parameters)
             unsafe
             {
                 if (parameters.Exponent == null || parameters.Modulus == null)
-                    throw new ArgumentException(SR.Cryptography_InvalidRsaParameters);
+                    throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
 
                 bool includePrivate = parameters.P != null && parameters.Q != null;
 
+                if (includePrivate && parameters.DP == null)
```

Comment:
Errm… probably because the null-DP case was the only one we were testing. We’re only writing out P and Q, so strictly speaking, there’s no reason to enforce rules on any of DQ/DQ/D/IQ – but maybe we should be enforcing them anyway to be consistent with OpenSSL/Csp. Does a null-check on each of these sound like a reasonably coherent behavior wrt the RSA family?